src/main/java/com/mixpanel/android/mpmetrics/ServerMessage.java

```diff
@@ -148,3 +155,20 @@ public boolean isOnline(Context context) {
 
     private static final String LOGTAG = "MixpanelAPI.ServerMessage";
 }
+
+class ServiceUnavailableException extends Exception {
+    public ServiceUnavailableException(String message, String strRetryAfter) {
+        super(message);
+        try {
+            mRetryAfter = Integer.parseInt(strRetryAfter);
+        } catch (NumberFormatException e) {
+            mRetryAfter = 0;
+        }
+    }
+
+    public int getRetryAfter() {
+        return mRetryAfter;
+    }
+
+    private int mRetryAfter;
```

Comment:
I actually made it a final in the very beginning.. but the java syntax checker says "Variable might already have been assigned to" at line 165.